### PR TITLE
Expand side bar automatically

### PR DESCRIPTION
**Pull request by rjvelazco**

### CODE REVIEW

File: components/SideNav.tsx [hunk 1]
```diff
@@ -4,21 +4,35 @@ import classNames from 'classnames';
...
         showSubList: null,
         active: null
     });
-
+    useEffect(() => {
+        setNavItem({
+            showSubList: breadCrumb[level],
+            active: breadCrumb[level]
+        });
+    }, [breadCrumb]);
```
Comment (fmontes) [inline components/SideNav.tsx:35]: Why useEffect?

File: pages/_app.tsx [hunk 2]
```diff
@@ -30,8 +32,15 @@ const HeaderWrapper = styled.div`
...
     const pageTitle = pageProps.pageTitle || 'Documentation';
+    const navData = pageProps.navDot ? pageProps.navDot[0] : [];
     const [showSidebar, setShowSidebar] = useState(true);
     const [showSideToc, setShowSideToc] = useState(false);
+    const [breadCrumb, setBreadcrumb] = useState([]);
+    const router = useRouter();
+    const docPage = router.asPath.split('/')[2];
+    useEffect(() => {
+        setBreadcrumb(searchBreadCrump(navData.dotcmsdocumentationchildren, docPage));
+    }, []);
```
Comment (fmontes) [inline pages/_app.tsx:43]: Why `useEffect`?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: pages/_app.tsx (hunk 2 above)
Comment (rjvelazco) [reply pages/_app.tsx:43]: To make sure the `setBreadCrumb` just runs once, the first time the page is open. we only care that the `sidenav` appears open on the first page the user visits.

File: components/SideNav.tsx (hunk 1 above)
Comment (rjvelazco) [reply components/SideNav.tsx:35]: We set the nav item when the `breadCrumb` changes which will change once.

**Pull request merged**